### ISSUE TITLE
Allow "never" in rule default-case

### ISSUE THREAD

**Issue by Tiseno** [eslint/eslint#15682]
### What rule do you want to change?

default-case

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
/*eslint default-case: ["error", "never"]*/

switch (a) {
    case 1:
        /* code */
        break;

    default:
        /* code */
        break;
}
```

### What does the rule currently do for this code?

Right now there is only an option to enforce that default cases must be used in switch statements, but no option to enforce that default should not be used at all.
In my experience default cases bring more trouble because it indicates that some cases have not been handled.
Also, it breaks type checking where the type system can detect if the switch case is exhaustive or not. For example by using [switch-exhaustiveness-check](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/switch-exhaustiveness-check.md)

### What will the rule do after it's changed?

The rule will be able to be configured to not only enforce usage of default cases, but also to enforce not to use default cases.

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @Tiseno, thanks for the issue!

The purpose of  `default-case` rule is to warn of those `switch` statements in which the omission of `default:` may have been unintentional (the rule requires a `// no default` comment to indicate intention), so I'm not sure if an option that disallows `default:` makes sense for this rule.

You can disallow `default:` by configuring the [no-restricted-syntax](https://eslint.org/docs/rules/no-restricted-syntax) rule with a `SwitchCase[test=null]` selector - [Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCB7XG5cdHNlbGVjdG9yOiBcIlN3aXRjaENhc2VbdGVzdD1udWxsXVwiLFxuICAgIG1lc3NhZ2U6IFwiRG9uJ3QgdXNlIGBkZWZhdWx0OmAuXCJcbn1dKi9cblxuc3dpdGNoIChmb28pIHtcbiAgY2FzZSBiYXI6XG4gICAgYmF6KCk7XG4gICAgYnJlYWs7XG4gICAgXG4gIGRlZmF1bHQ6IC8vIGVycm9yXG4gICAgcXV1eCgpO1xufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=).

**Comment by Tiseno** [comment]
I was thinking about a no-default-case rule instead, but that would go against guide lines to not having two rules that contradict each other. This change is consistent with a lot of other rules where one can choose to enforce or disable a feature, however. That this rule accepts a special comment syntax to disable the check seems more like a bug to me than a feature. But I guess that feature could still be present even with this change no?

Nice! no-restricted-syntax is good enough for me, tyvm

**Comment by mdjermanovic** [comment]
> This change is consistent with a lot of other rules where one can choose to enforce or disable a feature, however. That this rule accepts a special comment syntax to disable the check seems more like a bug to me than a feature.

Rules that accept special comments have a different purpose than rules that require/disallow some feature based on configuration. The purpose of rules that accept special comments is to enforce indicating the intention to use or not use a feature in places where using or not using that feature looks like a possible error, not to entirely require/disallow the feature.

`default-case` treats missing `default` cases as possible errors and requires a special comment to indicate intention.

Similarly, `no-fallthrough` treats fallthroughs between cases as possible errors, and requires a special comment to indicate that the fallthrough was intentional.

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.

**Comment by Tiseno** [comment]
Alright, cool, thanks, I will try custom rule as well.
Just curious, what rules are classified as not stylistic?
The readme doesn't specify what a stylistic rule is.
I would argue to disable default cases has to do with correctness, is that stylistic?
